libs/stats/odc/stats/_cli_run_gm.py

```diff
@@ -56,8 +65,15 @@ def run_gm(cache_file, tasks, dryrun, verbose, threads, overwrite, public, locat
         print(repr(rdr))
 
     def _proc(task):
-        ds_in = gm_input_data(task, resampling=resampling)
-        ds = gm_reduce(ds_in)
+        NY, NX = task.geobox.shape
+
+        ds_in = gm_input_data(task, resampling=resampling, chunk=(NY//y_chunks, NX))
+        tdim = list(ds_in.dims)[0]
+        ds_in = ds_in.chunk({tdim: -1, 'x': NX//x_chunks})
```

Comment:
I'm not sure what tdim: -1 achieves.

Reply:
combines all time slices into one column, `-1` means make chunk as big as the whole dimension.

Reply:
since geomedian needs to look at the whole time dimension at once and we use simple `map_blocks` to achieve it we need rechunk that way, `int_geomedian` can do that internally, but it might be best to rechunk once rather many times, rechunking is comparatively expensive operation even when operating within one single worker, and much worse on a larger cluster.